src/datastore_mad/remotes/libfs.sh

```diff
@@ -410,7 +414,11 @@ function fs_size {
             unlink "${IMAGE}" 2>/dev/null
         fi
     else
-        echo 'File not found'
+        # Print to stdout the error message generated by `ls`, to distinguish for example between
+        # non-existent file or permission issue. Example:
+        #   Output from ls: "ls: cannot access 'image.raw': Permission denied"
+        #   After sed: "cannot access 'image.raw' (Permission denied)"
+        ls "$SRC" 2>&1 | sed -e 's/^ls: //' -e 's/: \(.*\)/ (\1)/'
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using &#39;ls&#39; for file operations in shell scripts can be problematic, potentially leading to incorrect processing of filenames with spaces or special characters. This may result in data loss or unintended consequences. To mitigate this, use the more robust &#39;find&#39; command instead. &#39;find&#39; offers flexible options for locating and processing files safely. For more on secure coding practices, see OWASP&#39;s Secure Coding Practices Quick Reference Guide: https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/